src/datachain/lib/listing.py

```diff
@@ -1,19 +1,31 @@
 import asyncio
+import glob
+import os
 from collections.abc import AsyncIterator, Iterator, Sequence
-from typing import Callable, Optional
+from datetime import datetime, timedelta, timezone
+from typing import TYPE_CHECKING, Callable, Optional
 
 from botocore.exceptions import ClientError
 from fsspec.asyn import get_loop
+from sqlalchemy.sql.expression import true
 
 from datachain.asyn import iter_over_async
 from datachain.client import Client
 from datachain.error import ClientError as DataChainClientError
 from datachain.lib.file import File
+from datachain.query.schema import Column
+from datachain.sql.functions import path as pathfunc
+
+if TYPE_CHECKING:
+    from datachain.lib.dc import DataChain
+
 
 ResultQueue = asyncio.Queue[Optional[Sequence[File]]]
 
 DELIMITER = "/"  # Path delimiter
 FETCH_WORKERS = 100
+LISTING_TTL = 4 * 60 * 60  # cached listing lasts 4 days
+LISTING_PREFIX = "lst__"  # listing datasets start with this name
```

Comment:
We have `LISTING_PREFIX` in three places for now:
- `src/datachain/dataset.py`
- `src/datachain/lib/dc.py`
- and here
Just wanted to be sure we don't want to have all these prefix constants in one place 👀

Reply:
I've removed them from `dc.py` and about `src/datachain/dataset.py` - they are there because of legacy and backward compatibility and will be removed in one of the followup PRs (I've added a TODO where it's used)